homeassistant/components/aidot/coordinator.py

```diff
@@ -0,0 +1,168 @@
+        """Update data async."""
+        try:
+            status = await self.device_client.read_status()
+        except AidotNotLogin:
+            await self.device_client.async_login()
+            return DeviceStatusData()
+        return status
+
+    @property
+    def device_info(self) -> DeviceInformation:
+        """Device information."""
+        return self.device_client.info
+
+
+class AidotDeviceManagerCoordinator(
+    DataUpdateCoordinator[dict[str, AidotDeviceUpdateCoordinator]]
+):
+    """Class to manage fetching Aidot data."""
+
+    config_entry: AidotConfigEntry
+    client: AidotClient
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: AidotConfigEntry,
+    ) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            config_entry=config_entry,
+            name=DOMAIN,
+            update_interval=timedelta(hours=6),
+        )
+        self.client = AidotClient(
+            session=async_get_clientsession(hass),
+            token=config_entry.data[CONF_LOGIN_INFO],
+        )
+        self.client.start_discover()
+        self.client.set_token_fresh_cb(self.token_fresh_cb)
+        self.identifier = config_entry.entry_id
+
+    async def _async_setup(self) -> None:
+        """Set up the coordinator.
+
+        Can be overwritten by integrations to load data or resources
+        only once during the first refresh.
+        """
+        try:
+            await self.async_auto_login()
+        except AidotUserOrPassIncorrect as error:
+            raise ConfigEntryError from error
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Update data async."""
+        try:
+            data = await self.client.async_get_all_device()
+        except AidotAuthFailed as error:
+            self.token_fresh_cb()
+            raise ConfigEntryError from error
+
+        current_coordinators: dict[str, AidotDeviceUpdateCoordinator] = self.data
+        if current_coordinators is None:
+            current_coordinators = {}
+        final_coordinators: dict[str, AidotDeviceUpdateCoordinator] = {}
+        for device in data.get(CONF_DEVICE_LIST):
+            if (
+                device[CONF_TYPE] == Platform.LIGHT
+                and CONF_AES_KEY in device
+                and device[CONF_AES_KEY][0] is not None
+            ):
+                dev_id = device.get(CONF_ID)
+                update_coordinator = current_coordinators.get(dev_id)
+                device_client = self.client.get_device_client(device)
+                if update_coordinator is None:
+                    update_coordinator = AidotDeviceUpdateCoordinator(
+                        self.hass, self.config_entry, device_client
+                    )
+                    await update_coordinator.async_config_entry_first_refresh()
+                final_coordinators[dev_id] = update_coordinator
+        return final_coordinators
```

Comment:
```suggestion
                current_coordinators[dev_id] = update_coordinator
```
maybe rename `current_coordinators` to `device_coordinators` for clarity